sernet.gs.server/src/sernet/verinice/service/ConfigurationService.java

```diff
@@ -274,19 +285,72 @@ public String getName(String user) {
         }
         return result;
     }
-    
+
+    @Override
+    public boolean isWriteAllowed(CnATreeElement cte) {
+        // Server implementation of CnAElementHome.isWriteAllowed
+        try {
+            // Short cut: If no permission handling is needed than all objects
+            // are
+            // writable.
+            if (!authService.isPermissionHandlingNeeded()) {
+                return true;
+            }
+            // Short cut 2: If we are the admin, then everything is writable as
+            // well.
+            if (AuthenticationHelper.getInstance().currentUserHasRole(new String[] { ApplicationRoles.ROLE_ADMIN })) {
+                return true;
+            }
+            Set<String> userRoles = loadRoles();
+            for (Permission p : cte.getPermissions()) {
+                if (p != null && p.isWriteAllowed() && userRoles.contains(p.getRole())) {
+                    return true;
+                }
+            }
+        } catch (SecurityException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Write is not allowed", e);
+            }
+            return false;
+        } catch (sernet.gs.service.SecurityException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Write is not allowed", e);
+            }
+            return false;
+        } catch (RuntimeException re) {
+            LOG.error("Error while checking write permissions", re);
+            throw re;
+        } catch (CommandException t) {
+            LOG.error("Error while checking write permissions", t);
+            throw new RuntimeException("Error while checking write permissions", t);
+        }
+        return false;
+    }
+
+    public Set<String> loadRoles() throws CommandException {
```

Comment:
And we get circular reference between AccountService and ConfigurationService:

https://docs.spring.io/spring/docs/3.0.x/javadoc-api/org/springframework/beans/factory/FactoryBeanNotInitializedException.html